test/runnable/test17684.d

```diff
@@ -0,0 +1,12 @@
+private struct StaticStruct
+{
+    static int value;
+    static alias value this;
+}
+
+void main()
+{
+    StaticStruct = 42;
+    immutable int a = StaticStruct;
+    assert(a == 42);
+}
```

Comment:
Missing final end of line. Consider setting up [EditorConfig](http://editorconfig.org) on your text editor, so that repo text editor settings - https://github.com/dlang/dmd/blob/master/.editorconfig - are automatically applied.